### PR TITLE
Expanded subsystem activation tests

### PR DESCRIPTION
**Pull request by honza-kasik**
Expanded subsystem activation tests (added web.xml tests and negative test)

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [ ] N/A Link to the passing job is provided
- [x] Code is self-descriptive and/or documented
- [x] Description of the tests scenarios is included (see #46)

### CODE REVIEW

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.NonJwtJaxRsTestApplication;
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.SecuredJaxRsEndpoint;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+public class ActivationPropertiesUnsetTest {
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: I was wondering whether a line of Javadoc could be added for this class too, specifying that it is covering test cases for missing subsystem activation. The name could then be related more easily to the scope of the tests this way, WDYT?

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    private static final String NO_VALUE_DEPLOYMENT_NAME = "PROPERTIES_UNSET_DEPLOYMENT";
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: I didn't understand the names for this constants at first sight. Could something like `UNSET_AUTH_ACTIVATION_PROPERTY_DEPLOYMENT_NAME` be more clear in your opinion?

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    private static final String OTHER_VALUE_DEPLOYMENT_NAME = "OTHER_VALUE_DEPLOYMENT";
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: Same here, maybe something like `NON_JWT_AUTH_ACTIVATION_PROPERTY_DEPLOYMENT_NAME` could do, but this one of mine is just a suggestion.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    @Deployment(name = NO_VALUE_DEPLOYMENT_NAME, managed = false)
+    public static Archive<?> createDeploymentNoValue() {
+        return ShrinkWrap
+                .create(WebArchive.class,
+                        ActivationPropertiesUnsetTest.class.getSimpleName() + NO_VALUE_DEPLOYMENT_NAME + ".war")
+                .addClass(SecuredJaxRsEndpoint.class)
+                .addClass(NonJwtJaxRsTestApplication.class)
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: Would you evaluate whether to join the two `addClass()` statements, here and for all following `ShrinkWrap` archives generation? We've been following this practice in recent PRs.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/AnnotationActivationCompatibilityTest.java [hunk 5]
```diff
@@ -27,13 +27,14 @@
  * Just verify a raw token value can be injected into a variable in an application class a compare it has expected value
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/AnnotationActivationCompatibilityTest.java]: Maybe I don't understand the wording, but perhaps you meant that "a compare" to be "and compare"? 
Just guessing. Anyway the class name is about activating the subsystem via annotations, which is fine, so maybe this is what should be mentioned in this line. The test method description, later, is covering the specific case requirements already, IMO.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/WebXmlActivationCompatibilityTest.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.restassured.RestAssured;
+
+@RunWith(Arquillian.class)
+public class WebXmlActivationCompatibilityTest {
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/WebXmlActivationCompatibilityTest.java]: See my previous comment about the class Javadoc.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/publickeylocation/PublicKeyFileLocationTestCase.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.jboss.arquillian.container.test.api.OperateOnDeployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunAsClient
+@RunWith(Arquillian.class)
+public class PublicKeyFileLocationTestCase {
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/publickeylocation/PublicKeyFileLocationTestCase.java]: Why is this class here? Maybe future TD? Or is the PR still WIP?

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java (hunk 1 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: Thank you! Good point. I will add some JavaDoc since I am out of ideas for better name.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java (hunk 2 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: +1 Thank you.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java (hunk 3 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: +1, will edit.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/AnnotationActivationCompatibilityTest.java (hunk 5 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/AnnotationActivationCompatibilityTest.java]: Thanks, will edit.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/publickeylocation/PublicKeyFileLocationTestCase.java (hunk 7 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/publickeylocation/PublicKeyFileLocationTestCase.java]: Will remove. Nice catch!

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java (hunk 4 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: I am not sure I follow. Can you please provide me with an example?

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/WebXmlActivationCompatibilityTest.java (hunk 6 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/WebXmlActivationCompatibilityTest.java]: javaDoc added, thanks.

**Comment by honza-kasik** [comment]
Thank you @fabiobrz  I just pushed an update.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java (hunk 4 above)
Comment (fabiobrz) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: Sure:
```
.addClass(SecuredJaxRsEndpoint.class, NonJwtJaxRsTestApplication.class)
```

Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: To be honest, I like it better the way I did it first. Though if you want me to change it I have no objections...

**Comment by honza-kasik** [comment]
Removed file, fixed formatting.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.NonJwtJaxRsTestApplication;
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.SecuredJaxRsEndpoint;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Set of negative tests fo subsystem activation.
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: "fo" -> "for"

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java (hunk 4 above)
Comment (fabiobrz) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: I definitely wouldn't force you to do so :)
I was just trying and letting you know that, in a recent review, I was asked to do so and thus it seemed reasonable to me to keep the same code approach. 
Anyway, if you have some rationale for it (e.g. to emphasize the difference in using a non-JWT annotated class for JAX-RS application, I am fine with that. I'll leave this decision to you.

Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: Thank you, I understand why are pointing this out. To be honest, I have no objective arguments supporting my decision or arguments against my decision regarding using multiple method calls or one method call with multiple arguments. Maybe one (and this is just my POV) is that calling multiple methods makes it look more tidy and clear. Though there are probably other arguments which I'm failing to see.

If you are not against it, I would leave this as-is...

**Comment by honza-kasik** [comment]
Very good catch Fabio! Thank you! I fixed the JavaDoc and also test names seemed little odd to me. Pushing changes now.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+import org.junit.runner.RunWith;
+
+/**
+ * Set of negative tests for subsystem activation.
+ */
+@RunWith(Arquillian.class)
+public class ActivationPropertiesUnsetTest {
+
+    private static final String NON_JWT_AUTH_ACTIVATION_PROPERTY_DEPLOYMENT_NAME = "PROPERTIES_UNSET_DEPLOYMENT";
+    private static final String UNSET_AUTH_ACTIVATION_PROPERTY_DEPLOYMENT_NAME = "OTHER_VALUE_DEPLOYMENT";
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: Sorry if I am missing something but it seems to me that during the first round of review these two constants were switched.
What I mean: 

* `UNSET_AUTH_ACTIVATION_PROPERTY_DEPLOYMENT_NAME` - i.e. no authentication is activated because all related properties are unset  => "PROPERTIES_UNSET_DEPLOYMENT"

* `NON_JWT_AUTH_ACTIVATION_PROPERTY_DEPLOYMENT_NAME` - i.e. authentication is activated through some different (non JWT) mechanism => "OTHER_VALUE_DEPLOYMENT"

If the names I suggested shouldn't look clear to you, please feel free to choose something else, e.g. I see you used very clear names in other tests, as for instance `DEPLOYMENT_WITH_VALID_KEY`.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java (hunk 4 above)
Comment (fabiobrz) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: Thanks. As I stated earlier, the reasoning behind my comment was to have the same common approach to coding style for methods building a ShrinkWrap archive. 
That said, your reasoning about code being more tidy and clean makes sense to me too in this very case.
So let's agree on leaving this as-is.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    /**
+     * @tpTestDetails Test that subsystem is not activated when there is not {@code MP-JWT} set as an
+     *                {@code auth-method} anywhere (both web.xml and {@code @LoginConfig} annotation).
+     * @tpPassCrit Deployment fails because classes required by deployment (MP-JWT API) are not loaded due to the fact
+     *             that the subsystem was not activated.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test(expected = org.jboss.arquillian.container.spi.client.container.DeploymentException.class)
+    public void testDeploymentFailsWithUnsetAuth(@ArquillianResource Deployer deployer) {
+        deployer.deploy(NON_JWT_AUTH_ACTIVATION_PROPERTY_DEPLOYMENT_NAME);
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: Yes definitely it seems to me there is something wrong with those constants' name. Please see my comment on their declaration.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    /**
+     * @tpTestDetails Test that subsystem is not activated when the {@code auth-method} is set to other value than
+     *                {@code MP-JWT}.
+     * @tpPassCrit Deployment fails because classes required by deployment (MP-JWT API) are not loaded due to the fact
+     *             that the subsystem was not activated.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test(expected = org.jboss.arquillian.container.spi.client.container.DeploymentException.class)
+    public void testDeploymentFailsAuthOtherValue(@ArquillianResource Deployer deployer) {
+        deployer.deploy(UNSET_AUTH_ACTIVATION_PROPERTY_DEPLOYMENT_NAME);
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: ... Same here.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/AnnotationActivationCompatibilityTest.java [hunk 12]
```diff
@@ -24,16 +24,18 @@
 import io.restassured.RestAssured;
 
 /**
- * Just verify a raw token value can be injected into a variable in an application class a compare it has expected value
+ * Set of test which are verifying that {@code @LoginConfig} annotation works as it should - which activates the JWT
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/AnnotationActivationCompatibilityTest.java]: `Set of test` -> `Set of tests`
`... - which activates...` -> `... - i.e. activating`

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/WebXmlActivationCompatibilityTest.java [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.restassured.RestAssured;
+
+/**
+ * Set of test which are verifying that JWT subsystem can be activated by configuring {@code auth-method} in app's
```
Comment (fabiobrz) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/WebXmlActivationCompatibilityTest.java]: `Set of test` -> `Set of tests`

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java (hunk 9 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: Done, thanks.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java (hunk 10 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: Done

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java (hunk 11 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/ActivationPropertiesUnsetTest.java]: Done

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/AnnotationActivationCompatibilityTest.java (hunk 12 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/activation/AnnotationActivationCompatibilityTest.java]: Done, thanks

**Comment by honza-kasik** [comment]
Thanks, pushed another update.

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/publickeylocation/PublicKeyPropertyTestCase.java [hunk 14]
```diff
@@ -108,15 +110,15 @@ public void testJwkVerificationFailsWithInvalidKey(@ArquillianResource URL url)
     }
 
     /**
-     * @tpTestDetails A negative scenario where request with proper JWT is sent on server which has configured bad
-     *                public key. The server verifies the signature and authorizes user.
+     * @tpTestDetails Proper JWT is sent on server which has configured matching public key. The server verifies the
+     *                signature and authorizes user.
      * @tpPassCrit Token which was sent on server is sent back to client in response.
      * @tpSince EAP 7.3.0.CD19
```
Comment (mnovak1) [inline microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/publickeylocation/PublicKeyPropertyTestCase.java]: Should be 7.4.0.CD19

File: mp-ts.sh [hunk 15]
```diff
@@ -48,4 +48,4 @@ clean)
    echo "Usage: $0 {dist-zip|dist-run}"
 esac
 
-exit 0
```
Comment (mnovak1) [inline mp-ts.sh]: I'm no sure what was fixed here. Shouldn't it be part of different PR?

Comment (honza-kasik) [reply mp-ts.sh]: Nothing was fixed, but my IDE wont let me "fix" it and return back those trailing spaces :)

Comment (honza-kasik) [reply mp-ts.sh]: I'll try to squash commits, make git reset and commit without changes to this file...

File: microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/publickeylocation/PublicKeyPropertyTestCase.java (hunk 14 above)
Comment (honza-kasik) [reply microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/publickeylocation/PublicKeyPropertyTestCase.java]: Thanks, will fix.

**Comment by honza-kasik** [comment]
@mnovak1 Thanks, git magic done...

**Comment by mnovak1** [comment]
@honza-kasik thanks I one last request. Could you take a look at JWT tests and sync formatting when calling RestAssured, for example in:
https://github.com/jboss-eap-qe/eap-microprofile-test-suite/blob/fd3d784180f62f9b5e9de4638555f8596e0e9d2e/microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/publickeylocation/PublicKeyPropertyTestCase.java

is different in each test. I prefer the multi-line version :-)

I'll merge this PR but I expect that you create new one with those formatting fixes. thanks
